nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/ResultSetRecordSet.java

```diff
@@ -354,6 +358,7 @@ private static RecordFieldType getFieldType(final int sqlType, final String valu
             case Types.DATE:
                 return RecordFieldType.DATE;
             case Types.DECIMAL:
+                return RecordFieldType.BIGDECIMAL;
             case Types.DOUBLE:
             case Types.NUMERIC:
```

Comment:
The `NUMERIC` SQL type is a fixed-point decimal too.
I think it should be handled together with `DECIMAL` and return BigDecimal here.